#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

https://invent.kde.org/frameworks/breeze-icons/-/merge_requests/517

From 88bc82090e3e6a0869f62737d043fdf7bede6211 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ingo=20Kl=C3=B6cker?= <dev@ingo-kloecker.de>
Date: Wed, 19 Feb 2025 16:24:59 +0100
Subject: [PATCH] Allow to disable building of the icon library

For cross-builds we only need the tools.
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 225aadc4d..6026a931f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,36 +62,40 @@ option(SKIP_INSTALL_ICONS "Skip installing the icons files" OFF)
 # FIXME: remove after e.g. Kexi is ported
 option(BINARY_ICONS_RESOURCE "Install Qt binary resource file containing breeze icons (breeze-icons.rcc), should only be used if needed for compatibility" OFF)

-set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF6BreezeIcons")
-
-configure_package_config_file(
-  "${CMAKE_CURRENT_SOURCE_DIR}/KF6BreezeIconsConfig.cmake.in"
-  "${CMAKE_CURRENT_BINARY_DIR}/KF6BreezeIconsConfig.cmake"
-  INSTALL_DESTINATION  ${CMAKECONFIG_INSTALL_DIR}
-  PATH_VARS  CMAKE_INSTALL_PREFIX
-)
-
-install(FILES
-  "${CMAKE_CURRENT_BINARY_DIR}/KF6BreezeIconsConfig.cmake"
-  "${CMAKE_CURRENT_BINARY_DIR}/KF6BreezeIconsConfigVersion.cmake"
-  DESTINATION "${CMAKECONFIG_INSTALL_DIR}"
-  COMPONENT Devel
-)
-
-ecm_setup_version(PROJECT
-  VARIABLE_PREFIX BREEZEICONS
-  VERSION_HEADER ${CMAKE_CURRENT_BINARY_DIR}/breezeicons_version.h
-  PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KF6BreezeIconsConfigVersion.cmake"
-  SOVERSION 6)
-
-install(EXPORT KF6BreezeIconsTargets
-  DESTINATION "${CMAKECONFIG_INSTALL_DIR}"
-  FILE KF6BreezeIconsTargets.cmake
-  NAMESPACE KF6::
-)
-
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/breezeicons_version.h
-  DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF}/BreezeIcons COMPONENT Devel)
+option(WITH_ICONS_LIBRARY "Build the Breeze icons library" ON)
+
+if(WITH_ICONS_LIBRARY)
+    set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF6BreezeIcons")
+
+    configure_package_config_file(
+      "${CMAKE_CURRENT_SOURCE_DIR}/KF6BreezeIconsConfig.cmake.in"
+      "${CMAKE_CURRENT_BINARY_DIR}/KF6BreezeIconsConfig.cmake"
+      INSTALL_DESTINATION  ${CMAKECONFIG_INSTALL_DIR}
+      PATH_VARS  CMAKE_INSTALL_PREFIX
+    )
+
+    install(FILES
+      "${CMAKE_CURRENT_BINARY_DIR}/KF6BreezeIconsConfig.cmake"
+      "${CMAKE_CURRENT_BINARY_DIR}/KF6BreezeIconsConfigVersion.cmake"
+      DESTINATION "${CMAKECONFIG_INSTALL_DIR}"
+      COMPONENT Devel
+    )
+
+    ecm_setup_version(PROJECT
+      VARIABLE_PREFIX BREEZEICONS
+      VERSION_HEADER ${CMAKE_CURRENT_BINARY_DIR}/breezeicons_version.h
+      PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KF6BreezeIconsConfigVersion.cmake"
+      SOVERSION 6)
+
+    install(EXPORT KF6BreezeIconsTargets
+      DESTINATION "${CMAKECONFIG_INSTALL_DIR}"
+      FILE KF6BreezeIconsTargets.cmake
+      NAMESPACE KF6::
+    )
+
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/breezeicons_version.h
+      DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF}/BreezeIcons COMPONENT Devel)
+endif()

 # tools to generate icons and resources
 add_subdirectory(tools)
diff --git a/icons/CMakeLists.txt b/icons/CMakeLists.txt
index 6952c2b06..a2d32a833 100644
--- a/icons/CMakeLists.txt
+++ b/icons/CMakeLists.txt
@@ -91,57 +91,61 @@ if(NOT SKIP_INSTALL_ICONS)
     gtk_update_icon_cache(${BREEZE_DARK_INSTALL_DIR})
 endif()

-# helper functions to ensure we load the icon theme
-set(kbreezeicons_SRCS breezeicons.cpp)
-
-# use big resource variant, else we OOM on 32 bit machines
-# this has issues with LTO, we have a unit test that tells if stuff got lost in the DLL
-qt_add_big_resources(kbreezeicons_resource_SRCS
-    ${RESOURCE_FILE}
-    OPTIONS --root /icons/breeze
-)
-
-add_library(KF6BreezeIcons ${kbreezeicons_resource_SRCS} ${kbreezeicons_SRCS})
-add_library(KF6::BreezeIcons ALIAS KF6BreezeIcons)
-
-set_target_properties(KF6BreezeIcons
-    PROPERTIES
-        VERSION ${BREEZEICONS_VERSION}
-        SOVERSION ${BREEZEICONS_SOVERSION}
-        EXPORT_NAME BreezeIcons
-)
-
-generate_export_header(KF6BreezeIcons BASE_NAME BreezeIcons)
-
-target_include_directories(KF6BreezeIcons
-    INTERFACE
-        "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF}/BreezeIcons>"
-)
-
-target_link_libraries(KF6BreezeIcons
-    PUBLIC
-        Qt6::Gui
-)
-
-ecm_generate_headers(KF6BreezeIcons_CamelCase_HEADERS
-    HEADER_NAMES
-        BreezeIcons
-
-    REQUIRED_HEADERS
-        KF6BreezeIcons_HEADERS
-)
-
-install(TARGETS KF6BreezeIcons EXPORT KF6BreezeIconsTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
-
-install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/breezeicons_export.h
-    ${KF6BreezeIcons_HEADERS}
-    DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF}/BreezeIcons
-    COMPONENT Devel
-)
-
-install(FILES
-    ${KF6BreezeIcons_CamelCase_HEADERS}
-    DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF}/BreezeIcons
-    COMPONENT Devel
-)
+if (WITH_ICONS_LIBRARY)
+    # helper functions to ensure we load the icon theme
+    set(kbreezeicons_SRCS breezeicons.cpp)
+
+    # use big resource variant, else we OOM on 32 bit machines
+    # this has issues with LTO, we have a unit test that tells if stuff got lost in the DLL
+    qt_add_big_resources(kbreezeicons_resource_SRCS
+        ${RESOURCE_FILE}
+        OPTIONS --root /icons/breeze
+    )
+
+
+    add_library(KF6BreezeIcons ${kbreezeicons_resource_SRCS} ${kbreezeicons_SRCS})
+    add_library(KF6::BreezeIcons ALIAS KF6BreezeIcons)
+
+    set_target_properties(KF6BreezeIcons
+        PROPERTIES
+            VERSION ${BREEZEICONS_VERSION}
+            SOVERSION ${BREEZEICONS_SOVERSION}
+            EXPORT_NAME BreezeIcons
+    )
+
+    generate_export_header(KF6BreezeIcons BASE_NAME BreezeIcons)
+
+    target_include_directories(KF6BreezeIcons
+        INTERFACE
+            "$<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR_KF}/BreezeIcons>"
+    )
+
+    target_link_libraries(KF6BreezeIcons
+        PUBLIC
+            Qt6::Gui
+    )
+
+    ecm_generate_headers(KF6BreezeIcons_CamelCase_HEADERS
+        HEADER_NAMES
+            BreezeIcons
+
+        REQUIRED_HEADERS
+            KF6BreezeIcons_HEADERS
+    )
+
+    install(TARGETS KF6BreezeIcons EXPORT KF6BreezeIconsTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
+
+    install(FILES
+        ${CMAKE_CURRENT_BINARY_DIR}/breezeicons_export.h
+        ${KF6BreezeIcons_HEADERS}
+        DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF}/BreezeIcons
+        COMPONENT Devel
+    )
+
+    install(FILES
+        ${KF6BreezeIcons_CamelCase_HEADERS}
+        DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF}/BreezeIcons
+        COMPONENT Devel
+    )
+
+endif()
